scripts/merge_truth_per_tract.py

```diff
@@ -162,8 +163,15 @@ def match_object_with_merged_truth(
     # Calculate magnitude difference for all the pairs
     with np.errstate(invalid="ignore"):
         dmag = np.abs(
-            object_cat.loc[object_idx, mag_label_obj].values -
-            truth_cat.loc[truth_idx, mag_label_truth].values
+            np.where(
```

Comment:
My preference would be to split this into two lines to make it easier to document what the first quantity is
```
mag_to_match = np.where( ... )
dmag = mag_to_match - truth_cat.loc[truth_idx, mag_label_truth].values
del mag_to_match
```

It don't think that there's a memory performance difference here in terms of temporary arrays.  I think the `np.where` command is going to generate the same temporary array anyway.